### PR TITLE
[release/7.0] Fix canceling flush of CancelInvocation when canceling server-to-client stream

### LINKED ISSUE #46861: [SignalR] Server doesn't stop streaming after cancellation on client
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

While experimenting with SignalR server-to-client streaming, I noticed that the server does not stop streaming if it was canceled on the client side.
The client continue to receive stream items and just ignore it (https://gist.github.com/AndreiGorlov/6925f610f880ff212bf146f6b85e965d#file-client-log-L119-L128)
The server receives an unexpected CancelInvocationMessage after the stream has been completely transferred (https://gist.github.com/AndreiGorlov/7710d3a590eb55295e9912bf830bdc39#file-server-log-L47-L48)

### Expected Behavior

Server stops streaming after cancellation.

### Steps To Reproduce

repro: https://github.com/AndreiGorlov/SignalRStreamSandbox
server.log: https://gist.github.com/AndreiGorlov/7710d3a590eb55295e9912bf830bdc39
client.log: https://gist.github.com/AndreiGorlov/6925f610f880ff212bf146f6b85e965d

### Exceptions (if any)

_No response_

### .NET Version

7.0.200

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by BrennanConroy**
# Fix canceling flush of CancelInvocation when canceling server-to-client stream

## Description

With server-to-client streaming, the client can cancel the stream and tell the server to stop sending more stream items. This cancellation isn't working properly in 7.0 anymore.

Fixes #46861

## Customer Impact

The server keeps doing work to produce stream items even though the client will ignore them. This creates unnecessary work (cpu and network) on both the server and client that can be avoided.

## Regression?

- [x] Yes
- [ ] No

Regressed by #41180 in 7.0

## Risk

- [ ] High
- [ ] Medium
- [x] Low

Reverted the part of the 7.0 change that regressed this, and added a test for the missing coverage

## Verification

- [x] Manual (required)
- [x] Automated

## Packaging changes reviewed?

- [ ] Yes
- [ ] No
- [x] N/A

### CODE REVIEW

**Comment by wtgodbe** [comment]
I can merge once this has PR approval

**Review by JamesNK — APPROVED** [review]

**Pull request merged**
